File: examples/developer-hub-go/flare/fetch_anchor_feeds.go
// THIS IS EXAMPLE CODE. DO NOT USE THIS CODE IN PRODUCTION.
package flare

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

const (
	BaseURL = "https://flr-data-availability.flare.network/"
	ApiKey  = "<your-api-key>"
)

var FeedIds = []string{
	"0x01464c522f55534400000000000000000000000000", // FLR/USD
	"0x014254432f55534400000000000000000000000000", // BTC/USD
	"0x014554482f55534400000000000000000000000000", // ETH/USD
}
var VotingRoundId = ""

type AnchorFeed struct {
	Body struct {
		VotingRoundID int    `json:"votingRoundId"`
		ID            string `json:"id"`
		Value         int    `json:"value"`
		TurnoutBIPS   int    `json:"turnoutBIPS"`
		Decimals      int    `json:"decimals"`
	} `json:"body"`
	Proof []string `json:"proof"`
}

func FetchAnchorFeeds() ([]AnchorFeed, error) {
	url := BaseURL + "api/v0/ftso/anchor-feeds-with-proof"
	if VotingRoundId != "" {
		url += "?voting_round_id=" + VotingRoundId
	}

	payload, _ := json.Marshal(map[string]interface{}{"feed_ids": FeedIds})
	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(payload))
	req.Header.Set("X-API-KEY", ApiKey)
	req.Header.Set("Content-Type", "application/json")

	resp, err := http.DefaultClient.Do(req)
	if err != nil || resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("request failed: %v (status: %d)", err, resp.StatusCode)
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, fmt.Errorf("failed to read response: %w", err)
	}

	var feeds []AnchorFeed
	return feeds, json.Unmarshal(body, &feeds)
}

func main() {
	feeds, err := FetchAnchorFeeds()
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}
	fmt.Printf("Anchor feeds: %+v\n", feeds)
}

File: examples/developer-hub-rust/src/bin/fetch_anchor_feeds.rs
// THIS IS EXAMPLE CODE. DO NOT USE THIS CODE IN PRODUCTION.
#![allow(dead_code)]
#![allow(unused)]

use reqwest::Client;
use serde_json::json;

const BASE_URL: &str = "https://flr-data-availability.flare.network/";
const API_KEY: &str = "<your-api-key>";

const FEED_IDS: &[&str] = &[
    "0x01464c522f55534400000000000000000000000000", // FLR/USD
    "0x014254432f55534400000000000000000000000000", // BTC/USD
    "0x014554482f55534400000000000000000000000000", // ETH/USD
];

pub async fn fetch_anchor_feed(
    feed_ids: &[&str],
    voting_round_id: Option<u32>,
) -> Result<Vec<serde_json::Value>, reqwest::Error> {
    let client = Client::new();

    let mut url = format!("{BASE_URL}api/v0/ftso/anchor-feeds-with-proof");
    if let Some(id) = voting_round_id {
        url.push_str(&format!("?voting_round_id={}", id));
    }

    let response = client
        .post(&url)
        .header("X-API-KEY", API_KEY)
        .header("Content-Type", "application/json")
        .json(&json!({ "feed_ids": feed_ids }))
        .send()
        .await?;

    let json: Vec<serde_json::Value> = response.json().await?;
    Ok(json)
}

#[tokio::main]
async fn main() {
    match fetch_anchor_feed(FEED_IDS, None).await {
        Ok(data) => println!("Anchor feeds data: {:?}", data),
        Err(err) => eprintln!("Error fetching anchor feeds: {}", err),
    }
}
